operator/pkg/reconciliation/reconcile_racks.go

```diff
@@ -2137,6 +2155,10 @@ func (rc *ReconciliationContext) ReconcileAllRacks() (reconcile.Result, error) {
 		return recResult.Output()
 	}
 
+	if recResult := rc.DecommissionNodes(endpointData); recResult.Completed() {
```

Comment:
why not move this up before CheckRackPodTemplate?

Reply:
initially I put it after everything based on the idea of "we should make sure that everything else is in good shape before we attempt to decommission" but if that idea is fulfilled before we get to `CheckRackPodTemplate` then I'll move it up.

Reply:
If we put this anywhere after the `CheckCassandraNodeStatuses()`, we'll know that all the pods are up, running, and ready. That works for knowing the pods in the cluster are in a happy state. Where we put this call after that call tells you what operations might get priority over decomissioning, and those that may interrupt decomissioning. For example, if you put the call after `CheckRollingRestart()`, then in-between decomissioning two nodes, a rolling restart might happen.

Reply:
By the way, the call is probably here because of advice I gave earlier, but your right, it probably should be moved. I don't much care for how other datacenter-wide operations can get interleaved with each other but putting the call here doesn't actually solve that. Also, maybe interleaving is fine.